astropy/units/tests/test_quantity_array_methods.py

```diff
@@ -640,3 +640,20 @@ def test_assignment_with_different_names(self):
         assert qra[1].value == np.array((-1000.0, -2000.0, -3000.0), qra.dtype)
         # Ensure we do not override dtype names of value.
         assert value.dtype.names == ("x", "y", "z")
+
+
+class TestArrayAPI:
+    def setup_method(self):
+        import warnings
+
+        warnings.filterwarnings("ignore", category=UserWarning)
+        import numpy.array_api as np
+
+        self.array = np.asarray([1, 2, 3])
+
+    def test_array_api_init(self):
+        quantity_array = u.Quantity(self.array, u.m)
+        assert quantity_array.unit == u.m
```

Comment:
We have `astropy.tests.helper.assert_quantity_allclose()` function for such checks.